pydra/engine/specs.py

```diff
@@ -156,55 +156,69 @@ def check_fields_input_spec(self):
 
         """
         fields = attr_fields(self)
-        names = []
-        require_to_check = {}
-        for fld in fields:
-            mdata = fld.metadata
-            # checking if the mandatory field is provided
-            if getattr(self, fld.name) is attr.NOTHING:
-                if mdata.get("mandatory"):
-                    # checking if the mandatory field is provided elsewhere in the xor list
-                    in_exclusion_list = mdata.get("xor") is not None
-                    alreday_populated = in_exclusion_list and [
-                        getattr(self, el)
-                        for el in mdata["xor"]
-                        if (getattr(self, el) is not attr.NOTHING)
-                    ]
-                    if (
-                        alreday_populated
-                    ):  # another input satisfies mandatory attribute via xor condition
-                        continue
+
+        for field in fields:
+            field_is_mandatory = bool(field.metadata.get("mandatory"))
+            field_is_unset = getattr(self, field.name) is attr.NOTHING
```

Comment:
Optimization: Do not build alternatives if no problem exists.

```suggestion
            field_is_unset = getattr(self, field.name) is attr.NOTHING
            
            if not field_is_mandatory or not field_is_mandatory:
                continue
```

The `continue` on L198 does this after constructing `alternative_fields` and `required_fields`.